iocore/net/SSLUtils.cc

```diff
@@ -425,7 +425,7 @@ ssl_client_hello_callback(SSL *s, int *al, void *arg)
 {
   const char *servername = nullptr;
   const unsigned char *p;
-  size_t remaining, len;
+  size_t remaining = INT64_MAX, len;
   if (SSL_client_hello_get0_ext(s, TLSEXT_TYPE_server_name, &p, &remaining) || remaining <= 2) {
```

Comment:
Actually looking at this code again, I'm not sure what I was thinking with the or clause.  If there is no server_name extension present (SSL_client_hello_get0_ext returns 0), we shouldn't try to parse the data at all.   Perhaps I meant the remaining check to be an and?  I will ponder that a bit more.

Reply:
Looked back at thr test/handshaker_helper.c code I started from in the openssl repo.  I tried flipping the sense of some of the clauses, and I messed up this one.  I'll put up another PR with the sense of the clause fixed.